### DISCUSSION TITLE
Does it refresh the tokens when using IAM authentication?

### DISCUSSION THREAD

**Discussion by visa2learn** [aws/aws-advanced-jdbc-wrapper#271] [Q&A] [answered]
Hi,

I want to use aws-advanced-jdbc-driver for IAM authentication to RDS. Internally this wrapper must be generating the RDS token and use that as password for the database connection. The RDS token has 15 mins expiry time. So my question is whether this wrapper will refresh and get new token when the current one expires? It would help if you can point me to the code in this repo which does this refresh of tokens.

Thanks

**Comment by karenc-bq** [comment, marked as answer]
Hi @visa2learn, 
Yes, the AWS Advanced JDBC Driver will check and refresh the token whenever you establish a new connection.
https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/plugin/IamAuthConnectionPlugin.java#L125

**Reply by visa2learn** [reply]
Thanks @karenc-bq that helps
